### PR TITLE
Add 10 unit tests to readings.js

### LINKED ISSUE #698: improve data tests to do multiple values
### Is your feature request related to a problem? Please describe.

There are many places that OED tests that a certain value matches the expected value. While this is a good test, it would be better if it was done for multiple values. Examples include:

- src/server/test/routes/compressedReadingsRouteTests.js seems to test a single value.
- src/server/test/db/readingTests.js tests 1 or 2 points
- src/server/test/db/newCompressedReadingsTests.js tests limited points
- src/server/test/db/compareTests.js does a couple of points
- src/server/test/db/barchartReadingsTests.js does a few points

These tests also tend to have readings with 1 hour length.

### Describe the solution you'd like

OED now allows for readings of arbitrary length. Thus, tests should try reading lengths other than 1 hour. The automated test data (see src/server/data/automatedTestingData.js) includes 23 minute data for this reason. The other reason 23 minutes was chosen is that readings cross the hour and day boundaries so are better tests of the compression algorithms. We should be using automated tests to do this type of testing.

Since we can generate multiple points, it would be better to loop over a number of points and test each one in an automated fashion rather than doing a few specialized points.

The graphing is now done with compressed data that can be daily, hourly and original (raw) meter data. We should be testing the values returned from these to be sure they are correct (already done to some extent).

We should more generally test that groups properly add the included meters. The automated data for sine^2 + cosine^2 that gives a constant value is a good example of how this could be done.

### Describe alternatives you've considered

We can continue to test limited number of points and for easy time frames for those points and do the rest manually but that is not what we want.

### Additional context

None.

### PR DESCRIPTION
**Pull request by ZhenyuYu1**
# Description

This pull request adds 10 unit tests to readings.js based on the requirements from the [testing docs](https://github.com/OpenEnergyDashboard/DesignDocs/blob/main/testing/testing.md). 

- should have daily points for 15 minute reading intervals and flow units with +-inf start/end time & kW as kW
- 15 days barely gives hourly points & middle readings
- 14 days barely gives raw points & middle readings
- should have daily points for 15 minute reading intervals and quantity units with +-inf start/end time & kWh as MJ
- should have daily points for 15 minute reading intervals and quantity units with +-inf start/end time & kWh as MJ reverse conversion
- should have daily points for 15 minute reading intervals and quantity units with +-inf start/end time & kWh as BTU chained
- should have hourly points for middle readings of 15 minute for a 60 day period and quantity units & kWh as MJ
- should have hourly points for middle readings of 15 minute for a 60 day period and raw units & C as F with intercept
- should have raW points for middle readings of 15 minute for a 14 day period and quantity units & kWh as MJ
- should have raw points for middle readings of 15 minute for a 14 day period and raw units & C as F with intercept
- should have daily points for 15 minute reading intervals and flow units with +-inf start/end time & thing as thing where rate is 36

Fixes #698

## Type of change
- [ ] Note merging this changes the node modules
- [ ] Note merging this changes the database configuration.
- [ ] This change requires a documentation update

## Checklist

- [x] I have followed the [OED pull request](https://openenergydashboard.github.io/developer/pr.html) ideas
- [x] I have removed text in ( ) from the issue request

### CODE REVIEW

**Comment by ZhenyuYu1** [comment]
We worked off of development branch on our fork, but then read the guidelines and made a branch off of that to merge in. If that is an issue we can delete our commits off of dev branch on our fork.

**Comment by huss** [comment]
> We worked off of development branch on our fork, but then read the guidelines and made a branch off of that to merge in. If that is an issue we can delete our commits off of dev branch on our fork.

This should be fine. The issue is that if you do all your work off a branch then if you continue working/committing/pushing to that branch then it changes the PR. So, having separate branches allows for a developer to continue new work after a PR and before it is merged. The development branch on your own fork is not an issue except be aware that if you branch from it then it cannot be merged into OED until this PR is approved since it contains that work,

**Comment by huss** [comment]
Thanks to @iansw246 and @ZhenyuYu1 for their first PR. OED appreciates your contribution. From the records I see, neither of you have agreed to the [OED CLA](https://openenergydashboard.github.io/developer/cla.html). The project needs that before we can accept your contribution. Please let me know if you have any questions or if you think I missed your CLA. @AaronHamilton965 has a CLA record. Thanks.

File: src/server/test/web/readings.js
Comment (huss) [inline src/server/test/web/readings.js]: I think this test was already implmented (See starting line 273). That is why the design table indicated it was done.

File: src/server/test/web/readings.js
Comment (huss) [inline src/server/test/web/readings.js]: Would you be willing to fix my typo so raW -> raw?

**Review by huss — CHANGES_REQUESTED** [review]
Thank you for your first PR. I tried to carefully check all the added tests and everything seemed okay. I did make one comment that should be addressed and another to fix my own typo.

Just for the record, a few of the new data files are not yet used in any test. I think that is fine as they should be soon with more tests added.

I have updated the designDoc to reflect the tests done in the PR.

**Comment by huss** [comment]
The changes look fine. Thank you for doing them so quickly. This PR can be merged (and approved) once the final CLA is in place.

**Review by huss — APPROVED** [review]

**Comment by huss** [comment]
Thanks to @ZhenyuYu1 and @iansw246 for working on this so quickly. It is great and ready to merge.

**Pull request merged**
